### PR TITLE
Fix evdev axis reversal

### PR DESCRIPTION
**Pull request by kirbyfan64**
@toccata10 See if this works now.

### CODE REVIEW

**Comment by hcorion** [comment]
Please merge after #3283

**Comment by hcorion** [comment]
From what I can tell, this won't work if joystick squircling is on (and it's on by default).

**Comment by toccata10** [comment]
It's working with
Squircle Joysticks: false
in the config file

**Comment by toccata10** [comment]
Strangely, on Tekken 6, when I press down, it registers up (and up registers up). Seems to be the only game with this pb (with keyboard, it is ok).
[RPCS3-Tekken6.zip](https://github.com/RPCS3/rpcs3/files/1244354/RPCS3-Tekken6.zip)

**Comment by toccata10** [comment]
A new test with (line 474):
value = 255 - value;
It fixes tekken 6 and it's ok with the other games I tested.

File: rpcs3/evdev_joystick_handler.cpp [hunk 1]
```diff
@@ -460,13 +460,22 @@ void evdev_joystick_handler::thread_func()
...
                         }
                     }
                     else
-                        pad.m_sticks[axis].m_value = scale_axis(evt.code, evt.value);
+                    {
+                        int value = scale_axis(evt.code, evt.value);
+
+                        if (revaxis[axis])
+                        {
+                            value = 256 - value;
```
Comment (toccata10) [inline rpcs3/evdev_joystick_handler.cpp]: value = 255 - value; //fixes Tekken 6 input

**Comment by toccata10** [comment]
For Squircle:
With Reverse left stick X axis: true, line 451 should be
pad.m_sticks[Xaxis].m_value = scale_axis(ABS_X, 255-joy_axis[i][Xaxis]);

**Comment by toccata10** [comment]
Now that #3283 is merged, this fixes the missing reversal (without squircle):
`                     {
                        int value = scale_axis(evt.code, evt.value);

                        if (revaxis[axis])
                        {
                            value = 255 - value;
                        }

                        pad->m_sticks[axis].m_value = value;
                    }`

**Comment by toccata10** [comment]
@kirbyfan64 any chance to see this corrected so that it can be merged ? Thks

**Comment by toccata10** [comment]
For the squircle pb with axis reversal, this is working:
```
                if (g_evdev_joystick_config.squirclejoysticks)
                {
                        if (revaxis[axis])
						{
							joy_axis[i][axis] = 255-evt.value;
                        }
						else
						{
							joy_axis[i][axis] = evt.value;
						}
```
@kirbyfan64: do you want me to create a pull request ? I don't know if you're still interested (or simply have time) in this. That's my corrected file:
[evdev_joystick_handler-correct.zip](https://github.com/RPCS3/rpcs3/files/1326319/evdev_joystick_handler-correct.zip)

**Comment by kirbyfan64** [comment]
@toccata10 Sorry! I totally forgot about this. I pushed a new version that should work with both squircle and normal modes.

**Comment by toccata10** [comment]
Works fine now on my system. Thanks.

**Pull request merged**